kvdb-rocksdb/src/lib.rs

```diff
@@ -812,29 +819,35 @@ mod tests {
 
 	#[test]
 	fn secondary_db_get() -> io::Result<()> {
-		let tempdir = TempDir::new("")?;
+		let primary = TempDir::new("")?;
 		let config = DatabaseConfig::with_columns(1);
-		let db = Database::open(&config, tempdir.path().to_str().expect("tempdir path is valid unicode"))?;
+		let db = Database::open(&config, primary.path().to_str().expect("tempdir path is valid unicode"))?;
 
 		let key1 = b"key1";
 		let mut transaction = db.transaction();
 		transaction.put(0, key1, b"horse");
 		db.write(transaction)?;
 
-		let config = DatabaseConfig { secondary: true, ..DatabaseConfig::with_columns(1) };
-		let second_db = Database::open(&config, tempdir.path().to_str().expect("tempdir path is valid unicode"))?;
+		let config = DatabaseConfig {
+			secondary: TempDir::new("")?.path().to_str().map(|s| s.to_string()),
+			..DatabaseConfig::with_columns(1)
+		};
+		let second_db = Database::open(&config, primary.path().to_str().expect("tempdir path is valid unicode"))?;
 		assert_eq!(&*second_db.get(0, key1)?.unwrap(), b"horse");
 		Ok(())
 	}
 
 	#[test]
 	fn secondary_db_catch_up() -> io::Result<()> {
-		let tempdir = TempDir::new("")?;
+		let primary = TempDir::new("")?;
 		let config = DatabaseConfig::with_columns(1);
-		let db = Database::open(&config, tempdir.path().to_str().expect("tempdir path is valid unicode"))?;
-
-		let config = DatabaseConfig { secondary: true, ..DatabaseConfig::with_columns(1) };
-		let second_db = Database::open(&config, tempdir.path().to_str().expect("tempdir path is valid unicode"))?;
+		let db = Database::open(&config, primary.path().to_str().expect("tempdir path is valid unicode"))?;
+		db.try_catch_up_with_primary()?;
```

Comment:
So this should fail right?

Reply:
Yes! I actually just left that in by accident 

-- I'll make sure to remove it